Lib/webbrowser.py

```diff
@@ -15,14 +15,15 @@ class Error(Exception):
 
 _browsers = {}          # Dictionary of available browser controllers
 _tryorder = []          # Preference order of available browsers
+_preferred_browser = None # The preferred browser for the current environment
 
-def register(name, klass, instance=None, update_tryorder=1):
-    """Register a browser connector and, optionally, connection."""
+def register(name, klass, instance=None, priority=False):
+    """Register a browser connector."""
     _browsers[name.lower()] = [klass, instance]
-    if update_tryorder > 0:
-        _tryorder.append(name)
-    elif update_tryorder < 0:
+    if priority or (_preferred_browser and name in _preferred_browser):
```

Comment:
Note that 'name in _preferred_browsers' is not enough. xdg-settings returns e.g. 'firefox.desktop'.